src/Umbraco.Core/PropertyEditors/PropertyValueEditor.cs

```diff
@@ -248,7 +248,7 @@ public virtual object ConvertEditorToDb(ContentPropertyData editorValue, object
                 return null;
             }
 
-            var result = TryConvertValueToCrlType(editorValue.Value);
+            var result = TryConvertValueToCrlType((editorValue.Value != null && editorValue.Value.GetType() == typeof(JValue)) ? editorValue.Value.ToString() : editorValue.Value);

```

Comment:
Can we leave this line unchanged and instead fix `TryConvertValueToClrType`? It's already dealing with all sorts of conversions. Probably needs to start with something like
```
var jv = value as JValue;
if (jv != null) value = value.ToString();
```
This way, all the conversion magic remains in one place.